feat: Add `hideCalendarEventDetails` option to event types

joeauyeung opened the pull request:
Co-authored-by: Alex van Andel <emrysal@users.noreply.github.com>

## What does this PR do?

<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

This PR adds the option to event types to make the created calendar events on 3rd party calendars "private". This definition is different based on the calendar provider.

This functionality is enabled on the following calendars:
- Google calendar
- Outlook
- iCal / Apple calendar
- Generated ICS files in emails

This PR also adds "Don't show other attendees" to Outlook.

- Fixes #16815 (GitHub issue number)
- Fixes CAL-4410 (Linear issue number - should be visible at the bottom of the GitHub issue description)

<!-- Please provide a loom video for visual changes to speed up reviews
 Loom Video: https://www.loom.com/
-->

## Mandatory Tasks (DO NOT REMOVE)

- [ ] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration. Write details that help to start the tests -->

- For each calendar integration in this PR
	- Enable "Hide calendar event details" under the advanced tab in an event type
	- Create a booking
	- The 3rd party calendar event should be set as private as defined by the 3rd party integration
	- Disable this option in the event type
	- The 3rd party calendar event should be set at it's default visability

joeauyeung committed:
Add `hideCalendarEventDetails` to DB

Co-authored-by: Alex van Andel <emrysal@users.noreply.github.com>

joeauyeung committed:
Add option to hide calendar event details

joeauyeung committed:
Pass `hideCalendarEventDetails` to event object

joeauyeung committed:
Add to API

joeauyeung committed:
Add to CalendarEvent class

joeauyeung committed:
Pass `hideCalendarEventDetails` to calendar services

joeauyeung committed:
Adjust test

keithwillcode commented on apps/web/lib/event-types/[type]/getServerSideProps.ts:
```diff
@@ -40,6 +40,7 @@ export const getServerSideProps = async (context: GetServerSidePropsContext) =>
     await ssr.viewer.eventTypes.get.prefetch({ id: eventTypeId });
     try {
       const { eventType } = await ssr.viewer.eventTypes.get.fetch({ id: eventTypeId });
+      console.log("🚀 ~ getEventTypeById ~ eventType:", eventType);
```
Remove?

keithwillcode commented on apps/web/public/static/locales/en/common.json:
```diff
@@ -2613,5 +2613,7 @@
   "booking_limits_updated_successfully": "Booking limits updated successfully",
   "you_are_unauthorized_to_make_this_change_to_the_booking": "You are unauthorized to make this change to the booking",
+  "hide_calendar_event_details": "Hide calendar event details",
+  "description_hide_calendar_event_details": "Depending on your calendar, will set calendar event details to private",
```
This is pretty non-descriptive. Where do we plan to tell them which calendars this is supported on?

emrysal replied:
Agree this isn't the best approach, right now we will support ICS, iCalendar, Apple Calendar, Office 365 Calendar, Outlook Calendar, CalDav and Google Calendar. 

With that range of calendars the question is raised whether we want to mention "limited" calendar support at all.

SomayChauhan commented on packages/lib/CalendarService.ts:
```diff
@@ -162,6 +162,7 @@ export default abstract class BaseCalendarService implements Calendar {
          * [UPDATE]: Since we're not using the PUBLISH method to publish the iCalendar event and creating the event directly on iCal,
          * this shouldn't be an issue and we should be able to add attendees to the event right here.
          */
+        ...(calEventRaw.hideCalendarEventDetails ? { classification: "PRIVATE" } : {}),
```
this is throwing an error
![Screenshot 2024-09-26 at 11 18 22 AM](https://github.com/user-attachments/assets/5405c60f-ca25-410e-bff0-443bb524a249)

ibex088 committed:
fix: move `hideCalendarEventDetails` to `api/v2/event-types_2024_06_14` and remove from older versions

- Added `hideCalendarEventDetails` to `api/v2/event-types_2024_06_14`
- Removed `hideCalendarEventDetails` from `api/v1` and `api/v2/event-types_2024_04_15`

ibex088 committed:
Merge branch 'main' into private-calendar-events

emrysal committed:
fix: calEventRaw is undefined (use event instead)

emrysal committed:
chore: Remove debug artifact

emrysal committed:
fix: description id + update to copy

emrysal committed:
Merge branch 'main' into private-calendar-events

emrysal committed:
fix: Attempt at fixing type error

emrysal committed:
Merge remote-tracking branch 'refs/remotes/origin/private-calendar-events' into private-calendar-events

emrysal committed:
Add hideCalendarEventDetails to test builder

CarinaWolli commented on apps/web/public/static/locales/en/common.json:
```diff
@@ -2613,5 +2613,7 @@
   "limit_team_booking_frequency_description": "Limit how many times members can be booked across all team event types (collective and round-robin)",
   "booking_limits_updated_successfully": "Booking limits updated successfully",
   "you_are_unauthorized_to_make_this_change_to_the_booking": "You are unauthorized to make this change to the booking",
+  "hide_calendar_event_details": "Hide calendar event details",
```
Can we try to make this more descriptive as well?
'Make calendar event private'
'Hide calendar event details on shared calendars'

CarinaWolli commented on packages/features/bookings/lib/handleNewBooking.ts:
```diff
@@ -898,6 +898,7 @@ async function handler(
     conferenceCredentialId,
     destinationCalendar,
     hideCalendarNotes: eventType.hideCalendarNotes,
+    hideCalendarEventDetails: eventType.hideCalendarEventDetails,
```
We need to set `hideCalendarEventDetails` in other places as well. I'd say everywhere where we do `eventManager.create(evt)`, for example in `handleConfirmation`

CarinaWolli reviewed (changes requested):
It's not yet working for bookings that are not confirmed by default

emrysal committed:
Edit title copy of hide_calendar_event_details

CarinaWolli reviewed (dismissed):
As discussed with @emrysal, bookings that are not confirmed by default will be handled in a follow-up

ThyMinimalDev committed:
Merge branch 'main' into private-calendar-events

ThyMinimalDev committed:
fixup! Merge branch 'main' into private-calendar-events

emrysal reviewed (dismissed):
Happy with platform-libraries bump 👍

ThyMinimalDev committed:
fixup! fixup! Merge branch 'main' into private-calendar-events
